app/java/net/openid/appauthdemo/IdentityProvider.java

```diff
@@ -219,6 +220,12 @@ public void setClientId(String clientId) {
         mClientId = clientId;
     }
 
+    public String getClientSecret() {
+        return mClientSecret;
+    }
+
+    public void setClientSecret(String clientSecret) { mClientSecret = clientSecret; }
```

Comment:
I'm still generally unhappy about including client secrets in the demo app. I think it teaches people to do the wrong thing. At the very least, there should be some detailed commentary on these methods about using client secrets from non-confidential clients, such as I wrote here:

https://github.com/iainmcgin/AppAuth-Demo/blob/master/README.md